mediacapture-streams/GUM-required-constraint-with-ideal-value.https.html

```diff
@@ -0,0 +1,32 @@
+<!doctype html>
+<html>
+<head>
+<title>Ideal value in required constraint in getUserMedia</title>
+<link rel="author" title="Intel" href="http://www.intel.com"/>
+<link rel="help" href="https://w3c.github.io/mediacapture-main/#dfn-fitness-distance">
+</head>
+<body>
+<p class="instructions">When prompted, accept to share your video stream.</p>
+<h1 class="instructions">Description</h1>
+<p class="instructions">This test checks that setting a required constraint
+with an ideal value in getUserMedia works</p>
+
+<div id='log'></div>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script>
+var t = async_test("Tests that setting a required constraint with an ideal value in getUserMedia works");
+t.step(function() {
+  navigator.mediaDevices.getUserMedia({video: {width: {ideal: 320, min: 160}}})
+    .then(t.step_func(function (stream) {
+      assert_equals(stream.getVideoTracks().length, 1, "the media stream has exactly one video track");
+      assert_equals(stream.getVideoTracks()[0].getSettings().width, 320, 'ideal width is selected for getUserMedia() video tracks');
```

Comment:
Can you also add a video element and check video.videoWidth?
Something like:
video.srcObject = stream;
await video.play();
assert_equals(video.videoWidth, 320);

Reply:
Done.